### ISSUE TITLE
Support ansible.tower as well as awx.awx in roles

### ISSUE THREAD

**Issue by ericzolf** [redhat-cop/automate_tower_genie_collections#115]
I wasn't aware but you can [define a search list for collections in roles](https://docs.ansible.com/ansible/latest/user_guide/collections_using.html#using-collections-in-roles). That means that we should be able to remove the explicit `awx.awx` in all roles by adding something like the following to `meta/main.yml`:

```
collections:
- ansible.tower
- awx.awx
```
(assuming we want `ansible.tower` to be the default, and entries are ignored if they're not installed). That would make reuse in an enterprise context much easier.

What do you think?

**Comment by sean-m-sullivan** [comment]
I think its a good idea, but I'm wary of it using the internal modules that were a part of awx.awx that were packaged with 2.9, and can't remember how they did it for 2.10. So testing is needed. I believe we want to move off of calling them explicitly, and/or set minimum versions for each with each release.

**Comment by tonykay** [comment]
I would be cautious, and it is against the Ansible Recommendations. For example there are `roles` called `users`, `credentials` etc. As soon as you start using this collection in a much larger project where a user may do something like the below you have no idea where `users` is coming from:
```
collections:
    - foo.foo.
    - foo.bar
    - community.git
    - ansible.tower
    - awx.awx
```

One solution is to add context to `role` names `tower_configuration_users`

**Comment by ericzolf** [comment]
Perhaps one step back on what we're trying to reach: collections hosted by the CoP  should (IMHO always) be usable internally (IT. demos, agnosticd, etc), but also be usable by Consultants at customer so that they can deliver their work more efficiently and provide more value in the same time for our customers (and make it easy for them to provide back improvements). If internally, we don't care too much about support, we do at customer site, hence need to use ansible.tower and not awx.awx (our collection remains unsupported, no question).

If we all agree on the objectives, we can continue to discuss :smirk: I think that the renaming suggested by @tonykay is good, we should just chose a prefix not colliding (potentially at least) with the awx.awx/ansible.tower content, so `tower_` doesn't sound a good plan to me but I can't come with a better idea right now... something like `towerconf_users` perhaps?

**Comment by sean-m-sullivan** [comment]
Agree, we can rename them, and see about doing testing around this. I'd like to see it be agnostic, but could maybe even build the collection upload to do one or the other.

**Comment by sean-m-sullivan** [comment]
I just did testing on this, verifying that:
A. No default modules are included such as tower_label from the ansible install
B. That the module/role namespace/collection names were removed. 

Adding the following at the beginning of the file before tasks

```
  collections:
    - awx.awx
    - redhat_cop.tower_configuration
 ```
 Achieves this and allows us to use both. 
 
 My current branch is a little messy, and this will be a major change that we can implement. It is now planned, and I'll put in a PR here in the next week or two unless someone beats me to it.

**Comment by sean-m-sullivan** [comment]
Ok, after testing and some work I got this to work, It required putting putting the above codeblock in the meta/main.yml of each role, and it ran fine in playbooks, but failed in lint. This is a known issue for [ansible-lint. ](https://github.com/ansible-community/ansible-lint/issues/538). Will keep an eye on this, and keep the changes I made, however in the meantime will restore the fqdn for the module names.
